contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,207 @@
+
+A good rule of thumb to know if your change is too small is if it's unlikely to impact user behavior significantly. To help inform this decision, you can use qualitative data from user research or quantitative data from previous A/B tests.
+
+### 4. Has a sufficiently large sample size of users
+
+A large sample size for your test ensures the *statistical significance* of your experiment results i.e., if you can be confident that your results are not due to chance, but rather a true effect of the changes you made.
+
+To know whether your change is large enough, you'll need to know:
+
+1. Your current conversion rate.
+2. Your "minimum detectable effect" i.e., the smallest change in the conversion rate that you want to detect. The smaller the change, the larger the sample size you'll need.
+3. Your desired level of confidence (the industry standard is 95%).
+
+You then use a [formula](https://en.wikipedia.org/wiki/Sample_size_determination) to determine if your sample size is large enough. There are many calculators online that will do this for you, so you can avoid calculating this yourself (We also include this calculator when creating a new [A/B test in PostHog](/ab-testing/features)).
+
+### 5. Has a long enough test duration
+
+Building on the notion of ensuring statistical significant results, once you've calculated your required sample size, you'll be able to calculate how long your experiment should run for. You do this by dividing your sample size by your daily number of eligible users.
+
+For example, if you're making changes to your signup flow and your required sample size is 1,000 new sign-ups. If your daily number of sign-ups is 100, then you'll need to run your experiment for `1,000 / 100 = 10 days`.
+
+Once again, there are many calculators online to help you determine your duration, and PostHog also includes this calculator when creating a new experiment.
+
+**A good rule of thumb is that your experiment duration should be between one week and one month:
+**
+- One week is a good minimum since users may behave differently on weekends or weekdays.
+- One month is a good maximum since otherwise you may delay shipping important product changes.
+
+## When not to A/B test
+
+There's are costs and trade-offs to running A/B tests, and so it may not be a good idea to always run them. Here a few scenarios in which you may not want to run an A/B test:
+
+- **When you lack of sufficient traffic or users, or have time constraints:** It won't be possible to gather enough data to obtain the statistically significant results required to make good decisions.
+
+- **When you have high implementation costs:** Sometimes it may create a lot of technical debt to support and maintain both control and test variants of your experiment.
+
+- **When you have opportunity costs:** The time and resources it takes to implement and analyze A/B tests may be better spent working on other projects.
```

Comment:
These could be combined?

Reply:
Hmm perhaps. I think better to have them separate. Opprotunity cost is implied in "high implementation costs". but I think it's better to call it out as it's own point (it's not always necessarily related to implementation)